cmake/compiler_support.cmake

```diff
@@ -31,22 +31,65 @@ if (NOT PERL_FOUND )
    message(FATAL_ERROR "Perl not found")
 endif()
 
+
 set(MASM2GAS_PATH ${OMR_ROOT}/tools/compiler/scripts/masm2gas.pl CACHE INTERNAL "MASM2GAS PATH")
 
 include(${OMR_ROOT}/cmake/AddPrefix.cmake) 
 
-# Platform setup code!
-# TODO: THis needs to be abstracted for cross platform builds
+# Fetch the OMR view of the system.
+include(OmrDetectSystemInformation)
+
+macro(tr_detect_system_information)
+
+	macro(jit_not_ready)
+		message(FATAL "JIT isn't ready to build with CMake on this platform")
+	endmacro()
+
+
+	omr_detect_system_information()
+
+
+	set(TR_COMPILE_DEFINITIONS "")
+
+
+	# Platform setup code! 
+	# TODOs:
+	#  - Support more platforms, and, separate host and target arch. 
+	#  - Once we support all the platforms OMR supports with CMake, this can be
+	#    largely integrated into OmrDetectSystemInformation.cmake. 
+	if(OMR_ARCH_X86)
+		set(TR_HOST_ARCH    x     CACHE INTERNAL "The architecture directory used for the compiler code (x, p, or z)")
```

Comment:
I am not sure about all of the extra spaces to attempt to line things up.  Is this something the other CMake files are doing?  This type alignment usually ends up being broken and also causes a 1 line change to turn into much larger changes to fix alignment issues.